snapcraft/integrations/tests/test_travis.py

```diff
@@ -0,0 +1,185 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import subprocess
+from unittest import mock
+
+import fixtures
+import yaml
+
+from snapcraft import (
+    storeapi,
+    tests,
+)
+from snapcraft.integrations import travis
+
+
+test_snapcraft_yaml = """
+name: foo
+version: 1
+summary: a summary
+description: a description
+confinement: strict
+grade: stable
+parts:
+  foo:
+    plugin: nil
+"""
+
+
+class TravisTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.fake_logger = fixtures.FakeLogger(level=logging.DEBUG)
+        self.useFixture(self.fake_logger)
+        self.fake_terminal = tests.fixture_setup.FakeTerminal()
+        self.useFixture(self.fake_terminal)
+
+    def make_travis_yml(self, content):
+        with open('.travis.yml', 'w') as fd:
+            fd.write(content)
+
+    @mock.patch('subprocess.check_call')
+    def test_enable_missing_travis_cli(self, mock_check_call):
+        mock_check_call.side_effect = [FileNotFoundError()]
+
+        with self.assertRaises(EnvironmentError) as raised:
+            travis.enable()
+
+        self.assertEqual([
+            'Travis CLI (`travis`) is not available.',
+            'Please install it (e.g. `sudo gem install travis`) before '
+            'trying this command again.',
```

Comment:
At least on a vanilla lxc image, ruby isn't even installed, so this will just result in `sudo: gem: command not found`. Are we assuming the users will understand what a gem is and what they need to do in order to install them, or should we mention that ruby needs to be installed first?